Sources/VLCSettingsController.m

```diff
@@ -196,4 +209,43 @@ - (void)updateForPasscode:(NSString *)passcode
         [self.navigationController.presentedViewController dismissViewControllerAnimated:YES completion:nil];
     }
 }
+
+- (UITableViewCell*)newCellForIdentifier:(NSString*)identifier {
+    UITableViewCell *cell = nil;
+    if ([identifier isEqualToString:kIASKPSToggleSwitchSpecifier]) {
+        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:kIASKPSToggleSwitchSpecifier];
+        cell.accessoryView = [[IASKSwitch alloc] initWithFrame:CGRectMake(0, 0, 79, 27)];
+        [((IASKSwitch*)cell.accessoryView) addTarget:self action:@selector(toggledValue:) forControlEvents:UIControlEventValueChanged];
+        cell.selectionStyle = UITableViewCellSelectionStyleNone;
+    }
+    else if ([identifier isEqualToString:kIASKPSMultiValueSpecifier] || [identifier isEqualToString:kIASKPSTitleValueSpecifier]) {
+        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:identifier];
+        cell.accessoryType = [identifier isEqualToString:kIASKPSMultiValueSpecifier] ? UITableViewCellAccessoryDisclosureIndicator : UITableViewCellAccessoryNone;
+    }
+    else {
+        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:identifier];
+        if ([identifier isEqualToString:kIASKOpenURLSpecifier]) {
+            cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+        }
+    }
+    cell.textLabel.minimumScaleFactor = kIASKMinimumFontSize / cell.textLabel.font.pointSize;
+    cell.detailTextLabel.minimumScaleFactor = kIASKMinimumFontSize / cell.detailTextLabel.font.pointSize;
+    cell.textLabel.numberOfLines = 0;
+    cell.detailTextLabel.numberOfLines = 0;
+    
+    UIView *mask = [[UIView alloc] init];
+    mask.translatesAutoresizingMaskIntoConstraints = false;
+    [cell.contentView addSubview:mask];
+    
+    UILayoutGuide *margins = [cell.contentView layoutMarginsGuide];
+    [NSLayoutConstraint activateConstraints:@[
+                                              [mask.leadingAnchor constraintEqualToAnchor:margins.leadingAnchor],
+                                              [mask.trailingAnchor constraintEqualToAnchor:margins.trailingAnchor],
+                                              [mask.topAnchor constraintEqualToAnchor:margins.topAnchor],
+                                              [mask.bottomAnchor constraintEqualToAnchor:margins.bottomAnchor],
+                                              [mask.heightAnchor constraintGreaterThanOrEqualToConstant:MINIMUM_SETTINGS_ROW_HEIGHT]
```

Comment:
no hacks, please. If you want to ensure the correct height that leaves enough room you overwrite a cells systemLayoutSizeFittingSize or something like that. There are a bunch of articles on how to achieve this. Your implementation will not leave enough room when you have large text which is what you wanted if I understand the automatic dimension correctly. This will also make the MinimumSettingsrowheight obsolete